test/js/components/user/login.test.jsx

```diff
@@ -1,69 +1,30 @@
-import { fireEvent } from '@testing-library/react';
 import React from 'react';
 import { MemoryRouter, StaticRouter } from 'react-router-dom';
 
 import Login, { LoginButton } from '@/js/components/user/login';
-import { addUrlParams } from '@/js/utils/api';
 import routes from '@/js/utils/routes';
 
 import { renderWithRedux } from './../../utils';
 
 describe('<Login />', () => {
-  describe('login click', () => {
-    let location;
-
-    beforeAll(() => {
-      location = window.location;
-      delete window.location;
-      window.location = {
-        assign: jest.fn(),
-        pathname: location.pathname,
-        origin: location.origin,
-      };
-    });
-
-    afterAll(() => {
-      window.location = location;
-    });
-
-    test('updates `window.location.href` on login click', () => {
-      const { getByText } = renderWithRedux(
-        <MemoryRouter>
-          <Login />
-        </MemoryRouter>,
-      );
-      fireEvent.click(getByText('Log In With GitHub'));
-      const base = window.api_urls.github_login();
-      const expected = addUrlParams(base, { next: window.location.pathname });
-
-      expect(window.location.assign).toHaveBeenCalledWith(expected);
-    });
-
-    test('can use custom label and `from` path', () => {
-      const { getByText } = renderWithRedux(
-        <MemoryRouter>
-          <LoginButton label="Hi" from={{ pathname: '/custom' }} />
-        </MemoryRouter>,
-      );
-      fireEvent.click(getByText('Hi'));
-      const base = window.api_urls.github_login();
-      const expected = addUrlParams(base, { next: '/custom' });
-
-      expect(window.location.assign).toHaveBeenCalledWith(expected);
-    });
+  test('renders login form', () => {
+    const { getByText } = renderWithRedux(
+      <MemoryRouter>
+        <Login />
+      </MemoryRouter>,
+    );
+
+    expect(getByText('Log In With GitHub')).toBeVisible();
+  });
 
-    test('can use `location.state.from` path', () => {
-      const { getByText } = renderWithRedux(
-        <StaticRouter location={{ state: { from: { pathname: '/custom' } } }}>
-          <LoginButton />
-        </StaticRouter>,
-      );
-      fireEvent.click(getByText('Log In With GitHub'));
-      const base = window.api_urls.github_login();
-      const expected = addUrlParams(base, { next: '/custom' });
+  test('can use custom label and `from` path', () => {
+    const { getByText } = renderWithRedux(
+      <MemoryRouter>
+        <LoginButton label="Hi" from={{ pathname: '/custom' }} />
+      </MemoryRouter>,
+    );
 
-      expect(window.location.assign).toHaveBeenCalledWith(expected);
-    });
+    expect(getByText('Hi')).toBeVisible();
```

Comment:
I think in cases where we want to test custom props it might be worth asserting the value of the hidden inputs to ensure `next` is set as expected

Reply:
Good idea, I'll add that.

Reply:
Fixed in https://github.com/oddbird/Metecho/pull/1847/commits/4c65aedb20765730a178a7828ae5d95f173040d7